prime-router/src/main/kotlin/Submission.kt

```diff
@@ -76,6 +78,114 @@ class DetailedSubmissionHistory(
      */
     var externalName: String? = null
 
+    /**
+     * The step in the delivery process for a submission
+     * Supported values:
+     *     ERROR - error on initial submission
+     *     RECEIVED - passed the received step in the pipeline and awaits processing/routing
+     *     NOT_DELIVERING - processed but has no intended receivers
+     *     WAITING_TO_DELIVER - processed but yet to be sent to/downloaded by any receivers
+     *     PARTIALLY_DELIVERED - processed, successfully sent to/downloaded by at least one receiver
+     *     DELIVERED - processed, successfully sent to/downloaded by all receivers
+     * @todo For now, no "send error" type of state.
+     *     If a send error occurs for example,
+     *     it'll just sit in the waitingToDeliver or
+     *     partiallyDelivered state until someone fixes it.
+     */
+    enum class Status(val printableName: String) {
+        ERROR("Error"),
+        RECEIVED("Received"),
+        NOT_DELIVERING("Not Delivering"),
+        WAITING_TO_DELIVER("Waiting to Deliver"),
+        PARTIALLY_DELIVERED("Partially Delivered"),
+        DELIVERED("Delivered"),
+    }
+
+    /**
+     * Summary of how far along the submission's process is.
+     * The supported values arre listed in the Status enum.
+     */
+    val overallStatus: Status get() {
+        if (httpStatus != HttpStatus.OK.value() && httpStatus != HttpStatus.CREATED.value()) {
+            return Status.ERROR
+        }
+
+        if (destinations.size == 0) {
+            // 0 unfiltered destinations = unprocessed in almost every situation
+            return Status.RECEIVED
+        } else if (realDestinations.size == 0) {
+            return Status.NOT_DELIVERING
+        }
+
+        var finishedDestinations = 0
+
+        realDestinations.forEach {
+            var sentItemCount = 0
```

Comment:
_Currently_ we only send entire reports as complete chunks, no partial deliveries.  So for any one send or download, its sentItemCount will always == the destination's itemCount.  So _in theory_ you could dramatically simplify this code to this rule:
`if there is even one successful send or download, then that destination is done.`

OTOH, before you arrived, we used a delivery mechanism called 'Redox', that did partial deliveries - it took multiple partial delivery `send`s to deliver all the data.  The way you've written the code here might account for that, _as long as each successfully delivered item was only listed once in the numbers in the database_.